File: ant/src/com/appsecinc/ant/license/LICENSE_LicenseFile.java
package com.appsecinc.ant.license;

public class LICENSE_LicenseFile implements LicenseFile {

	@Override
	public String getExtension() {
		return "txt";
	}

	@Override
	public boolean isMatch(String filename) {
		String filenameLowerCase = filename.toLowerCase();
		return filenameLowerCase.equals("license")
			|| filenameLowerCase.equals("license.txt")
			|| filenameLowerCase.endsWith("_license.txt")
			|| filenameLowerCase.endsWith("_license")
			|| filenameLowerCase.endsWith("-license")
			|| filenameLowerCase.endsWith("-license.txt");
	}

	@Override
	public String getSpec(String filename) {
		String filenameLowerCase = filename.toLowerCase();
		filenameLowerCase = filenameLowerCase.replace(".txt", "");
		filenameLowerCase = filenameLowerCase.replace("-license", "");
		filenameLowerCase = filenameLowerCase.replace("_license", "");
		if (filenameLowerCase.equals("license")) return "";
		return filenameLowerCase;
	}

	@Override
	public String getType() {
		return null;
	}
}

File: msbuild/src/AppSecInc.LicensesCollector/LICENSE_LicenseFile.cs
using System;

namespace AppSecInc.LicensesCollector
{
    class LICENSE_LicenseFile : ILicenseFile
    {
        #region ILicenseFile Members

        public string GetExtension()
        {
            return "txt";
        }

        public bool IsMatch(string filename)
        {
            String filenameLowerCase = filename.ToLower();
            return filenameLowerCase.Equals("license")
                || filenameLowerCase.Equals("license.txt")
                || filenameLowerCase.EndsWith("_license.txt")
                || filenameLowerCase.EndsWith("_license")
                || filenameLowerCase.EndsWith("-license")
                || filenameLowerCase.EndsWith("-license.txt");
        }

        public string GetSpec(string filename)
        {
            String filenameLowerCase = filename.ToLower();
            filenameLowerCase = filenameLowerCase.Replace(".txt", "");
            filenameLowerCase = filenameLowerCase.Replace("-license", "");
            filenameLowerCase = filenameLowerCase.Replace("_license", "");
            if (filenameLowerCase.Equals("license")) return "";
            return filenameLowerCase;
        }

        public new string GetType()
        {
            return null;
        }

        #endregion
    }
}
